### ISSUE TITLE
[regression] `rm -i` prompt is failing (again)

### ISSUE THREAD

**Issue by mogenson** [xonsh/xonsh#3025]
### xonfig

<details>

```
$ xonfig
+------------------+-----------------+
| xonsh            | 0.8.10          |
| Python           | 3.7.2           |
| PLY              | 3.11            |
| have readline    | True            |
| prompt toolkit   | 2.0.8           |
| shell type       | prompt_toolkit2 |
| pygments         | 2.3.1           |
| on posix         | True            |
| on linux         | True            |
| distro           | arch            |
| on darwin        | False           |
| on windows       | False           |
| on cygwin        | False           |
| on msys2         | False           |
| is superuser     | False           |
| default encoding | utf-8           |
| xonsh encoding   | utf-8           |
| encoding errors  | surrogateescape |
+------------------+-----------------+
```

</details>

## issue

Looked through the old issues and it seems like this problem should be solved. So this might be a regression.

```
$ xonsh --no-rc
$ touch file1 file2 file3 file4
$ rm -i file*
/usr/bin/rm: remove regular empty file 'file1'? /usr/bin/rm: remove regular empty file 'file2'? /usr/bin/rm: remove regular empty file 'file3'? /usr/bin/rm: remove regular empty file 'file4'? /usr/bin/rm: error closing file
```

I checked that `rm` is not aliased to anything else in Bash. I also tried with `xonsh --no-rc --shell-type readline` with the same result. Finally, setting `$XONSH_STORE_STDOUT` to True or False makes no difference.

### versions

<details>

```
$ rm --version
rm (GNU coreutils) 8.30
$ uname -a
Linux 4.20.8-arch1-1-ARCH #1 SMP PREEMPT Wed Feb 13 00:06:58 UTC 2019 x86_64 GNU/Linux
```

</details>

**Comment by gforsyth** [comment]
Thanks for raising this, @mogenson!  I can confirm the regression and I'm pretty sure it's related to #3015 

Temporary workaround is to put the following in your `xonshrc`

```
__xonsh__.commands_cache.threadable_predictors['rm'] = lambda x: False
```

**Comment by scopatz** [comment]
Actually, doing a git bisect between 0.8.9 and 0.8.10 shows that it is e0aa871df71f24d951f726ed95eba6e91803d7ca which is the problem.  This is actually #3004.

**Comment by scopatz** [comment]
Potential fix in #3026

**Comment by scopatz** [comment]
Well, #3026 should still be merged, but I am seeing this as a random runtime failure.

**Comment by scopatz** [comment]
And actually I think this is more related to the fix in #3003
